pkg/api/validation/dynakube/oneagent.go

```diff
@@ -201,3 +208,53 @@ func isOneAgentVersionValid(_ context.Context, _ *Validator, dk *dynakube.DynaKu
 
 	return ""
 }
+
+func duplicateOneAgentArguments(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	args := dk.OneAgentArgumentsMap()
+	if args == nil {
+		return ""
+	}
+
+	for key, values := range args {
+		if key != "--set-host-property" && key != "--set-host-tag" && len(values) > 1 {
+			return fmt.Sprintf(errorDuplicateOneAgentArgument, key)
+		} else if key == "--set-host-tag" {
+			if duplicatedTags := findDuplicates(values); len(duplicatedTags) > 0 {
+				return fmt.Sprintf(errorSameHostTagMultipleTimes, duplicatedTags)
+			}
+		}
+	}
+
+	return ""
+}
+
+func forbiddenHostIdSourceArgument(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	args := dk.OneAgentArgumentsMap()
+	if args == nil {
+		return ""
+	}
+
+	for key := range args {
+		if dk.CloudNativeFullstackMode() && key == "--set-host-id-source" {
+			return errorHostIdSourceArgumentInCloudNative
+		}
+	}
+
+	return ""
+}
+
+func findDuplicates[S ~[]E, E comparable](s S) []E {
+	seen := make(map[E]bool)
+
+	duplicates := make([]E, 0)
+
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			seen[val] = true
+		} else if !slices.Contains(duplicates, val) {
+			duplicates = append(duplicates, val)
+		}
+	}
+
+	return duplicates
```

Comment:
A possible simplification
```suggestion 
	for _, val := range s {
		if _, ok := seen[val]; !ok {
			seen[val] = true
		} else {
			duplicates = append(duplicates, val)
		}
	}

	return slices.Compact(duplicates)
```